### DISCUSSION TITLE
Conditionally run a step only if a secret is defined?

### DISCUSSION THREAD

**Discussion by simonw** [community/community#25735] [Actions]
<p>I’d like to add a step to my workflow which only runs if a specific named secret has been defined.</p>
<p>I tried this but it didn’t work:</p>
<pre><code>if: secrets.MY_SECRET</code></pre>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi <a class="mention" href="/simonw">@simonw</a>,</p>
<p>Please use below format instead. <code>secrets.*</code> is not recognized in if expression. You can set it as env and read via <code>env.*</code>.</p>
<pre><code class="lang-auto">      - name: check
        if: contains(env.TESTKEY1, 'value1')         # use env value instead.
        run: |
          echo test
        env:
            TESTKEY1: ${{secrets.TESTKEY1}}         
</code></pre>
<p>Thanks</p>

**Comment by dtinth** [comment]
<p>To check if the secret which is converted into an <code>env</code> is defined, you can simply reference it in the <code>if</code> condition. For example:</p>
<p>For example:</p>
<pre><code class="lang-auto">      - name: SonarCloud Scan
        uses: SonarSource/sonarcloud-github-action@master
        if: env.SONAR_TOKEN
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }} # Needed to get PR information, if any
          SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
</code></pre>
